### DISCUSSION TITLE
Inventory item that generates / drops random items like a mob drop?

### DISCUSSION THREAD

**Discussion by WithinAmnesia** [flareteam/flare-game#855] [Q&A] [answered]
Inventory item that generates / drops random items like a mob drop? As in like with Old School RuneScape there are a few items that the player can obtain that have random items 'within' them that once 'opened' generate random amounts and or random types of items (with potentially random amounts) like a loot drop table of an enemy slain but in an inventory format. Example: https://oldschool.runescape.wiki/w/Casket + https://oldschool.runescape.wiki/w/Bird_nest + https://oldschool.runescape.wiki/w/Mystery_box . Is this feature possible to have featured within the Flare Engine? If so how can this be in game / how can with be made? Somehow so that an item can be used within the inventory that somehow exec reward_item... item drop table (is there even a command like that O.o)? Eh how would this play out in Flare, is it possible with the current build? I am kind of rolling the idea around in my head by I am kind of stuck as to how to pull it off XD.

**Comment by dorkster** [comment]
You can accomplish this with an event script. Here's an example of an item that drops 1-3 pieces of loot from Empyrean's level 1 loot table:

```
# scripts/test_script.txt
[event]
loot=loot/level_1.txt
loot_count=1,3
```

```
# items/items.txt
[item]
id=3000
name=Test item
INCLUDE items/base/misc/book.txt
quality=unique
script=scripts/test_script.txt
```

**Reply by WithinAmnesia** [reply]
This works? *Frantically testing*...

**Reply by dorkster** [reply]
Yes it works. You can also use `reward_loot` and `reward_loot_count` in the script file if you prefer the items to go directly into the player's inventory instead of dropping.

**Reply by WithinAmnesia** [reply]
Hmmmm *old internet router sounds* https://www.youtube.com/watch?v=QDS4B0mM-ew

**Reply by WithinAmnesia** [reply]
@dorkster I tested it and it kind of works yet it does not destroy itself after use (effectively making an infinite loot bug). How can this be made to consume the item / 'treasure trunk' when used to avoid infinite loot bug?
Icon 227 is a trunk I added: https://opengameart.org/content/flare-truck-icon
'#quality=fine' looks green and is a custom thing I added. I put in the item_type=consumable and script=scripts/test_script.txt to not crash on your end.
I later put item_type=consumable and max_quantity=1 and requires_item=3095 in hopes of making the item disappear after looking at consumable items / trying to copy how potions and scrolls work but I found no luck XD.

Here is the script (Item_Loot_Drop(Test)+.txt):
[event]
reward_loot=loot/level_1.txt
reward_loot_count=2,10


Here is the item (3095):
#Item Loot Drop Test (Mysterious Box)++
[item]
id=3095
name=Mysterious Sealed Trunk
#icon=227
icon=225 
level=10
#quality=fine
quality=one_time_use
item_type=consumable
soundfx=soundfx/inventory/inventory_wood.ogg
loot_animation=animations/loot/gem.txt
#script=scripts/AGABAL/Item_Loot_Drop(Test)+.txt
script=scripts/test_script.txt
flavor=Mysterious objects rattle inside when shaken.
max_quantity=1
requires_item=3095

**Reply by dorkster** [reply]
To make the item destroy itself, add `remove_item=3095` to the script file.

**Comment by WithinAmnesia** [comment, marked as answer]
I am going to try to add remove_item to script and a delay to avoid potential ~lag duping.

**Reply by WithinAmnesia** [reply]
@dorkster Success! Yet I swapped delay to cooldown and both don't seem to have an effect but the extra ~feature doesn't matter hopefully. It is quite fun, thanks Dorkster:-D!

Script:
[event]
reward_loot=loot/level_1.txt
reward_loot_count=2,10
cooldown=1 
remove_item=3096

Item:
#Item Loot Drop Test (Mysterious Box)+++
[item]
id=3096
name=Mysterious Sealed Trunk
icon=227
level=10
quality=fine
item_type=consumable
soundfx=soundfx/inventory/inventory_wood.ogg
loot_animation=animations/loot/gem.txt
script=scripts/AGABAL/Item_Loot_Drop(Test)+++.txt
flavor=Mysterious objects rattle inside when shaken.
max_quantity=1
requires_item=3096
